### PR TITLE
Case 21746: Fix color grading/lighting of renderLayer front entities

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.fogbugz.com/f/cases/21746/renderLayer-front-entities-rendering-dark-black

Test plan:
- Run [this](https://gist.githubusercontent.com/SamGondelman/1b7418c416df22ddc84683b72e23a2fd/raw/fc5f092e8ca728e2c7a78c73c751d6f936c13b58/RenderLayer.js).
- You'll see this:
image
- Press "o".  The colors/shading on most of the entity types will not change (the polyvox entity might change slightly).  The objects will draw in front now:
image
- Press "u".  The box, text, image, web, grid, and ring gizmo entities will become slightly transparent, but will still be lit the same way, and will still draw in front:
image

### CODE REVIEW

File: libraries/avatars-renderer/src/avatars-renderer/Avatar.cpp [hunk 1]
```diff
@@ -1661,60 +1661,6 @@ int Avatar::parseDataFromBuffer(const QByteArray& buffer) {
     return bytesRead;
 }
 
-int Avatar::_jointConesID = GeometryCache::UNKNOWN_ID;
-
-// render a makeshift cone section that serves as a body part connecting joint spheres
-void Avatar::renderJointConnectingCone(gpu::Batch& batch, glm::vec3 position1, glm::vec3 position2,
```
Comment (SamGondelman) [inline libraries/avatars-renderer/src/avatars-renderer/Avatar.cpp]: This method was inefficient and broken.

File: libraries/avatars-renderer/src/avatars-renderer/SkeletonModel.cpp [hunk 2]
```diff
@@ -338,24 +338,20 @@ void SkeletonModel::computeBoundingShape() {
...
     // draw a green cylinder between the two points
-    glm::vec3 origin(0.0f);
-    batch.setModelTransform(Transform().setTranslation(bottomPoint));
-    geometryCache->bindSimpleProgram(batch);
-    Avatar::renderJointConnectingCone(batch, origin, axis, scale * _boundingCapsuleRadius, scale * _boundingCapsuleRadius,
-                                      glm::vec4(0.6f, 0.8f, 0.6f, alpha));
+    float capsuleDiameter = 2.0f * _boundingCapsuleRadius;
+    glm::vec3 cylinderDimensions = glm::vec3(capsuleDiameter, _boundingCapsuleHeight, capsuleDiameter);
+    batch.setModelTransform(Transform().setScale(scale * cylinderDimensions).setRotation(_rotation).setTranslation(0.5f * (topPoint + bottomPoint)));
+    geometryCache->renderSolidShapeInstance(args, batch, GeometryCache::Shape::Cylinder, glm::vec4(0.6f, 0.8f, 0.6f, alpha));
```
Comment (SamGondelman) [inline libraries/avatars-renderer/src/avatars-renderer/SkeletonModel.cpp]: Now we just draw a cylinder, like we were trying to.

File: libraries/render-utils/src/GeometryCache.cpp [hunk 3]
```diff
@@ -1029,7 +1017,7 @@ void GeometryCache::updateVertices(int id, const QVector<glm::vec2>& points, con
     int* colorData = new int[details.vertices];
     int* colorDataAt = colorData;
 
-    const glm::vec3 NORMAL(0.0f, 0.0f, 1.0f);
+    const glm::vec3 NORMAL(0.0f, 1.0f, 0.0f);
```
Comment (SamGondelman) [inline libraries/render-utils/src/GeometryCache.cpp]: This fixes the normals of Ring Gizmo entities.  These methods aren't used anywhere else where the normals matter.

File: libraries/render-utils/src/GeometryCache.cpp [hunk 4]
```diff
@@ -2018,77 +2006,6 @@ void GeometryCache::renderLine(gpu::Batch& batch, const glm::vec2& p1, const glm
     batch.draw(gpu::LINES, 2, 0);
 }
 
-
-void GeometryCache::renderGlowLine(gpu::Batch& batch, const glm::vec3& p1, const glm::vec3& p2,
```
Comment (SamGondelman) [inline libraries/render-utils/src/GeometryCache.cpp]: unused

File: libraries/render-utils/src/GeometryCache.cpp [hunk 5]
```diff
@@ -2282,8 +2199,7 @@ gpu::PipelinePointer GeometryCache::getSimplePipeline(bool textured, bool transp
                 _unlitShader = _forwardUnlitShader;
             } else {
                 _simpleShader = gpu::Shader::createProgram(simple_textured);
-                // Use the forward pipeline for both here, otherwise transparents will be unlit
```
Comment (SamGondelman) [inline libraries/render-utils/src/GeometryCache.cpp]: the shader does the right thing now

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18187/HighFidelity-Beta-PR15197-8748a75.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18187/HighFidelity-Quest-Beta-PR15197-8748a75.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18187/HighFidelity-Beta-PR15197-8748a75.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18187/HighFidelity-Beta-Interface-PR15197-8748a75.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18194/HighFidelity-Beta-PR15197-e374be4.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18194/HighFidelity-Beta-Interface-PR15197-e374be4.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18194/HighFidelity-Beta-PR15197-e374be4.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18194/HighFidelity-Quest-Beta-PR15197-e374be4.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18194/HighFidelity-Beta-PR15197-e374be4.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18194/HighFidelity-Beta-Interface-PR15197-e374be4.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18198/HighFidelity-Beta-PR15197-79dda7a.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18198/HighFidelity-Quest-Beta-PR15197-79dda7a.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18198/HighFidelity-Beta-PR15197-79dda7a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18198/HighFidelity-Beta-Interface-PR15197-79dda7a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18198/HighFidelity-Beta-PR15197-79dda7a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18198/HighFidelity-Beta-Interface-PR15197-79dda7a.exe

**Pull request merged**